popx/migration_box.go

```diff
@@ -146,3 +203,24 @@ func (fm *MigrationBox) findMigrations(runner func([]byte) func(mf Migration, c
 		return nil
 	})
 }
+
+// hasDownMigrationWithVersion checks if there is a migration with the given
+// version.
+func (fm *MigrationBox) hasDownMigrationWithVersion(version string) bool {
```

Comment:
This might be a bit too strict because the following migration set is valid: {`a.mysql.up.sql`, `a.postgres.up.sql`, `a.down.sql`}. This code that I wrote in Hydra may or may not be helpful: https://github.com/ory/hydra/blob/d6080061e91844fa4ba67eea1110ec84cb20ab4b/cmd/cli/handler_migrate.go#L200